source/val/validate_decorations.cpp

```diff
@@ -848,9 +841,40 @@ spv_result_t CheckDecorationsOfEntryPoints(ValidationState_t& vstate) {
             break;
           if (auto error = CheckBuiltInVariable(interface, vstate))
             return error;
-        } else if (isBuiltInVar(interface, vstate)) {
-          if (auto error = CheckBuiltInVariable(interface, vstate))
-            return error;
+        } else {
+          // If there is not builtin block, check for builtin on the variable
+          for (auto& dec : vstate.id_decorations(interface)) {
+            if (dec.dec_type() != spv::Decoration::BuiltIn) continue;
+
+            if (auto error = CheckBuiltInVariable(interface, vstate))
+              return error;
+
+            if (!spvIsVulkanEnv(vstate.context()->target_env)) continue;
+
+            const spv::BuiltIn builtin = dec.builtin();
+            if (storage_class == spv::StorageClass::Input) {
```

Comment:
Might make sense to refactor this a bit if get used in the built-in structure checks.